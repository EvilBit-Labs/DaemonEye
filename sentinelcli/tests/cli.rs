use assert_cmd::prelude::*;
use insta::assert_snapshot;
use std::process::Command;

#[test]
fn prints_expected_greeting() -> Result<(), Box<dyn std::error::Error>> {
    use tempfile::tempdir;

    let temp_dir = tempdir()?;
    let db_path = temp_dir.path().join("test.db");

    let mut cmd = Command::cargo_bin("sentinelcli")?;
    if let Some(path_str) = db_path.to_str() {
        cmd.env("SENTINELCLI_DATABASE_PATH", path_str);
    } else {
        return Err("Invalid database path".into());
    }

    let output = cmd.output()?;
    assert!(output.status.success());
    let stdout = String::from_utf8_lossy(&output.stdout);
    assert_snapshot!("sentinelcli_greeting", stdout);
    Ok(())
}
