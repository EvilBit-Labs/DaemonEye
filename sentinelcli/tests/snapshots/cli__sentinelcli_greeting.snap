---
source: sentinelcli/tests/cli.rs
expression: stdout
---
SentinelD Database Statistics
============================
Processes: 0
Rules: 0
Alerts: 0
System Info: 0
Scans: 0
Health status: healthy
sentinelcli completed successfully
